File: interview-core/src/main/java/com/aims/core/question/Question.java
package com.aims.core.question;

import java.time.Instant;
import java.util.List;

/**
 * 题库领域模型。
 *
 * @param id 题目 ID
 * @param category 题目分类（对应 {@link QuestionCategory} 名称）
 * @param topic 主题
 * @param difficulty 难度（对应 {@link Difficulty} 名称）
 * @param content 题干
 * @param standardAnswer 标准答案
 * @param tags 标签
 * @param embedding 题目向量
 * @param createdAt 创建时间
 * @param updatedAt 更新时间
 */
public record Question(
        Long id,
        String category,
        String topic,
        String difficulty,
        String content,
        String standardAnswer,
        List<String> tags,
        String embedding,
        Instant createdAt,
        Instant updatedAt) {}

File: frontend/src/types/question.ts
/** 对齐 QuestionResponse */
export interface QuestionResponse {
  id: number;
  category: string;
  topic: string;
  difficulty: string;
  content: string;
  standardAnswer: string | null;
  tags: string[];
  hasEmbedding: boolean;
  createdAt: string;
  updatedAt: string;
}

/** 对齐 CreateQuestionRequest */
export interface CreateQuestionRequest {
  category: string;
  topic: string;
  difficulty: string;
  content: string;
  standardAnswer?: string;
  tags?: string[];
}

/** 对齐 UpdateQuestionRequest */
export interface UpdateQuestionRequest {
  category?: string;
  topic?: string;
  difficulty?: string;
  content?: string;
  standardAnswer?: string;
  tags?: string[];
}

/** 对齐 QuestionImportRequest */
export interface QuestionImportRequest {
  questions: CreateQuestionRequest[];
}

/** 对齐 ParsedQuestionResponse（面经解析结果，未落库） */
export interface ParsedQuestion {
  category: string;
  topic: string;
  difficulty: string;
  content: string;
  standardAnswer: string;
  tags: string[];
  matchedExistingId: number | null;
}

/** 对齐 InterviewNoteParseRequest */
export interface InterviewNoteParseRequest {
  text: string;
  categoryHint?: string;
}

/** 对齐 InterviewNoteParseTaskResponse（异步任务，提交/轮询共用） */
export interface InterviewNoteParseTask {
  taskId: string;
  status: string; // RUNNING / SUCCESS / FAILED / NOT_FOUND
  message: string | null;
  results: ParsedQuestion[] | null;
}

/** 对齐 QuestionSearchResult */
export interface QuestionSearchResult {
  id: number;
  category: string;
  topic: string;
  difficulty: string;
  content: string;
  standardAnswer: string | null;
  score: number;
}

/** 列表查询参数 */
export interface QuestionQuery {
  page?: number;
  size?: number;
  category?: string;
  difficulty?: string;
  topic?: string;
}
